src/VisualStudio/Core/Def/Implementation/LanguageClient/RazorLanguageClient.cs

```diff
@@ -0,0 +1,118 @@
+    [ClientName(ClientName)]
+    [Export(typeof(ILanguageClient))]
+    class RazorLanguageClient : ILanguageClient
+    {
+        private readonly IDiagnosticService _diagnosticService;
+        private readonly LanguageServerProtocol _languageServerProtocol;
+        private readonly Workspace _workspace;
+
+        public const string ClientName = "RazorCSharp";
+
+        /// <summary>
+        /// Gets the name of the language client (displayed to the user).
+        /// </summary>
+        public string Name => ServicesVSResources.Razor_CSharp_Language_Server_Client;
+
+        /// <summary>
+        /// Unused, implementing <see cref="ILanguageClient"/>
+        /// No additional settings are provided for this server, so we do not need any configuration section names.
+        /// </summary>
+        public IEnumerable<string>? ConfigurationSections { get; } = null;
+
+        /// <summary>
+        /// Gets the initialization options object the client wants to send when 'initialize' message is sent.
+        /// See https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/#initialize
+        /// We do not provide any additional initialization options.
+        /// </summary>
+        public object? InitializationOptions { get; } = null;
+
+        /// <summary>
+        /// Unused, implementing <see cref="ILanguageClient"/>
+        /// Files that we care about are already provided and watched by the workspace.
+        /// </summary>
+        public IEnumerable<string>? FilesToWatch { get; } = null;
+
+        public event AsyncEventHandler<EventArgs>? StartAsync;
+
+        /// <summary>
+        /// Unused, implementing <see cref="ILanguageClient"/>
+        /// </summary>
+        public event AsyncEventHandler<EventArgs>? StopAsync { add { } remove { } }
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public RazorLanguageClient(LanguageServerProtocol languageServerProtocol, VisualStudioWorkspace workspace, IDiagnosticService diagnosticService)
+        {
+            _languageServerProtocol = languageServerProtocol;
+            _workspace = workspace;
+            _diagnosticService = diagnosticService;
+        }
+
+        public Task<Connection> ActivateAsync(CancellationToken token)
+        {
+            var (clientStream, serverStream) = FullDuplexStream.CreatePair();
+            _ = new RazorLanguageServer(serverStream, serverStream, _languageServerProtocol, _workspace, _diagnosticService);
+            return Task.FromResult(new Connection(clientStream, clientStream));
+        }
+
+        /// <summary>
+        /// Signals that the extension has been loaded.  The server can be started immediately, or wait for user action to start.  
+        /// To start the server, invoke the <see cref="StartAsync"/> event;
+        /// </summary>
+        public async Task OnLoadedAsync()
```

Comment:
Did we want to have a base class for this and our regular LanguageClient so all of this can be reused? The base class would be abstract as a few things need to change, but at least it's a relatively small bit then.